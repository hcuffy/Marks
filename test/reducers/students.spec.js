/* eslint-disable max-len */
import applyStudentData from '../../app/reducers/students'
import {
	ADD_NEW_STUDENT,
	GET_ALL_STUDENTS,
	GET_SINGLE_STUDENT,
	DISPLAY_STUDENT_GRAPH,
	DISPLAY_STUDENT_SUBJECT_GRAPH
} from '../../app/components/students/constants'

const initialLoadState = {
	firstname: '',
	lastname: '',
	studentModal: false,
	studentDropdown: false,
	subjectDropdown: false,
	studentGraphId: null,
	chartToDisplay: null,
	studentGraphName: null,
	subjectGraphName: null
}

describe('test student reducer', () => {
	it('should use default state', () => {
		expect(applyStudentData(initialLoadState, {})).toMatchSnapshot()
	})

	it('should add new student', () => {
		expect(
			applyStudentData(initialLoadState, { type: ADD_NEW_STUDENT, payload: {} })
		).toMatchSnapshot()
	})

	it('should get all students', () => {
		const students = [
			{
				firstname: 'John',
				lastname: 'Tester',
				gender: 'male',
				classroom: 'Class 2',
				_id: 'EySxHRVO4693O78I',
				createdAt: { $$date: 1556364347040 },
				updatedAt: { $$date: 1556515640626 }
			}, {
				firstname: 'Jane',
				lastname: 'Tester',
				gender: 'female',
				classroom: 'Class 2',
				_id: 'zHDJF8nohVmf2Trurjh',
				createdAt: { $$date: 1556364347040 },
				updatedAt: { $$date: 1556515640626 }
			}
		]
		const payload = { students }
		expect(
			applyStudentData(initialLoadState, { type: GET_ALL_STUDENTS, payload })
		).toMatchSnapshot()
	})

	it('should get single student', () => {
		const payload = 'EySxHRVO4693O78I'
		expect(
			applyStudentData(initialLoadState, { type: GET_SINGLE_STUDENT, payload })
		).toMatchSnapshot()
	})

	it('should display student graph', () => {
		const payload = {
			studentGraphId: 'EySxHRVO4693O78I',
			studentGraphName: 'John Tester',
			chartToDisplay: 'student'
		}
		expect(
			applyStudentData(initialLoadState, {
				type: DISPLAY_STUDENT_GRAPH,
				payload
			})
		).toMatchSnapshot()
	})

	it("should display student's subject graph", () => {
		const payload = {
			subjectGraphId: 'qPtCHoVmf2Tbznji',
			subjectGraphName: 'Biology',
			chartToDisplay: 'subject'
		}

		expect(
			applyStudentData(initialLoadState, {
				type: DISPLAY_STUDENT_SUBJECT_GRAPH,
				payload
			})
		).toMatchSnapshot()
	})
})
