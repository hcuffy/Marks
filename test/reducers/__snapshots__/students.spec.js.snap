// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test student reducer should add new student 1`] = `
Object {
  "chartToDisplay": null,
  "firstname": "",
  "lastname": "",
  "studentDropdown": false,
  "studentGraphId": null,
  "studentGraphName": null,
  "studentModal": false,
  "subjectDropdown": false,
  "subjectGraphName": null,
}
`;

exports[`test student reducer should display student graph 1`] = `
Object {
  "chartToDisplay": "student",
  "firstname": "",
  "lastname": "",
  "studentDropdown": true,
  "studentGraphId": "EySxHRVO4693O78I",
  "studentGraphName": "John Tester",
  "studentModal": false,
  "subjectDropdown": false,
  "subjectGraphName": null,
}
`;

exports[`test student reducer should display student's subject graph 1`] = `
Object {
  "chartToDisplay": "subject",
  "firstname": "",
  "lastname": "",
  "studentDropdown": false,
  "studentGraphId": null,
  "studentGraphName": null,
  "studentModal": false,
  "subjectDropdown": true,
  "subjectGraphId": "qPtCHoVmf2Tbznji",
  "subjectGraphName": "Biology",
}
`;

exports[`test student reducer should get all students 1`] = `
Object {
  "chartToDisplay": null,
  "firstname": "",
  "lastname": "",
  "studentDropdown": false,
  "studentGraphId": null,
  "studentGraphName": null,
  "studentModal": false,
  "students": Array [
    Object {
      "_id": "EySxHRVO4693O78I",
      "classroom": "Class 2",
      "createdAt": Object {
        "$$date": 1556364347040,
      },
      "firstname": "John",
      "gender": "male",
      "lastname": "Tester",
      "updatedAt": Object {
        "$$date": 1556515640626,
      },
    },
    Object {
      "_id": "zHDJF8nohVmf2Trurjh",
      "classroom": "Class 2",
      "createdAt": Object {
        "$$date": 1556364347040,
      },
      "firstname": "Jane",
      "gender": "female",
      "lastname": "Tester",
      "updatedAt": Object {
        "$$date": 1556515640626,
      },
    },
  ],
  "subjectDropdown": false,
  "subjectGraphName": null,
}
`;

exports[`test student reducer should get single student 1`] = `
Object {
  "chartToDisplay": null,
  "firstname": "",
  "lastname": "",
  "studentDropdown": false,
  "studentGraphId": null,
  "studentGraphName": null,
  "studentId": "EySxHRVO4693O78I",
  "studentModal": true,
  "subjectDropdown": false,
  "subjectGraphName": null,
}
`;

exports[`test student reducer should use default state 1`] = `
Object {
  "chartToDisplay": null,
  "firstname": "",
  "lastname": "",
  "studentDropdown": false,
  "studentGraphId": null,
  "studentGraphName": null,
  "studentModal": false,
  "subjectDropdown": false,
  "subjectGraphName": null,
}
`;
